components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java

```diff
@@ -3349,9 +3386,20 @@ private User getSCIMUser(org.wso2.carbon.user.core.common.User coreUser, List<St
 
             // Add username with domain name.
             if (mandateDomainForUsernamesAndGroupNamesInResponse()) {
-                scimUser.setUserName(prependDomain(coreUser.getDomainQualifiedUsername()));
+                //If primary login identifire is enabled, set the username value of scim response to that value.
+                if (isLoginIdentifiresEnabled() && StringUtils.isNotEmpty(getPrimaryLoginIdentifireClaim())) {
```

Comment:
since `getPrimaryLoginIdentifireClaim()` method is getting called in multiple places, let's assign this value to a variable and call the variable wherever needed.